src/mesh/analysis.rs

```diff
@@ -227,19 +256,20 @@ pub fn is_mesh_watertight(edge_sharing: &EdgeSharingMap) -> bool {
 /// **must** be triangulated and watertight for this to produce usable results.
 ///
 /// The genus (G) is computed as: `V - E + F = 2*(1 - G)`.
-#[allow(dead_code)]
 pub fn triangulated_mesh_genus(vertex_count: usize, edge_count: usize, face_count: usize) -> i32 {
     1 - (cast_i32(vertex_count) - cast_i32(edge_count) + cast_i32(face_count)) / 2
 }
 
 /// Checks if two meshes are similar.
 ///
+/// This function is slow and is therefore enabled only for tests.
+///
 /// Two mesh geometries are similar when they are visually similar (see the
 /// definition of `are_visually_similar`), and they have the same number of
 /// vertices and normals. Therefore they are going to be treated the same by all
 /// functions of this software and all their transformations result in similar
 /// mesh geometries.
-#[allow(dead_code)]
+#[cfg(test)]
```

Comment:
No - this is to prevent us from using these functions in non-test builds. Notice I also added a comment that these functions can only be used in tests. They are now only compiled when running tests.

Reply:
we've got more such cases though. will we mark them test-only just when they are slow?

Reply:
Yes. We have a lot of dead code.

> will we mark them test-only just when they are slow?

We should mark them test only if they are only for tests. But yeah, we should try to not have slow code.

Reply:
Or at least I remember us discussing, these functions will only ever be used for tests some time in the past. If that's not true, I can remove the annotation.